### PR TITLE
Dev first party error handling

### PR DESCRIPTION
**Pull request by alexbpx**

### CODE REVIEW

**Comment by alexbpx** [comment]
Do not merge the PR if you approve, because kibana is down so I still need to validate the rest call.

File: PerimeterXModule/Internals/Validators/PXS2SValidator.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public bool VerifyS2S(PxContext PxContext)
 			}
 			catch (Exception ex)
 			{
-				Debug.WriteLine("Failed to verify S2S: " + ex.Message, PxConstants.LOG_CATEGORY);
+				PxLoggingUtils.LogError("Failed to verify S2S: " + ex.Message);
```
Comment (yaronpx) [inline PerimeterXModule/Internals/Validators/PXS2SValidator.cs]: debug instead of error

File: PerimeterXModule/PxModule.cs [hunk 2]
```diff
@@ -293,7 +294,7 @@ private void PostEnforcerTelemetryActivity()
 			}
 			catch (Exception ex)
 			{
-				Debug.WriteLine(string.Format("Encountered an error sending enforcer telemetry activity: {0}.", ex.Message), PxConstants.LOG_CATEGORY);
+				PxLoggingUtils.LogError(string.Format("Encountered an error sending enforcer telemetry activity: {0}.", ex.Message));
```
Comment (yaronpx) [inline PerimeterXModule/PxModule.cs]: debug instead of error

File: PerimeterXModule/Internals/ActivityReporter.cs [hunk 3]
```diff
@@ -63,13 +63,16 @@ public ActivityReporter(string baseUri, int capacity = 500, int bulkSize = 10, i
...
+			PxLoggingUtils.LogDebug("Reporter initialized");
         }
 
         public bool Post(Activity activity)
         {
             var added = activities.TryAdd(activity);
-            Debug.WriteLineIf(!added, "Failed to post activity", PxConstants.LOG_CATEGORY);
+			if (!added)
+			{
+				PxLoggingUtils.LogError("Failed to post activity");
```
Comment (yaronpx) [inline PerimeterXModule/Internals/ActivityReporter.cs]: debug instead of error

File: PerimeterXModule/Internals/ActivityReporter.cs [hunk 4]
```diff
@@ -108,12 +111,12 @@ private void SendActivities(List<Activity> activities)
...
                 var content = new StringContent(stringBuilder.ToString(), Encoding.UTF8, "application/json");
                 var response = httpClient.PostAsync(postUri, content).Result;
-                Debug.WriteLine(string.Format("Post request for {0} ({1}), returned {2}", postUri, stringBuilder.ToString(), response), PxConstants.LOG_CATEGORY);
+				PxLoggingUtils.LogDebug(string.Format("Post request for {0} ({1}), returned {2}", postUri, stringBuilder.ToString(), response));
                 response.EnsureSuccessStatusCode();
             }
             catch (Exception ex)
             {
-                Debug.WriteLine(string.Format("Failed sending activities (count {0}) - {1}", activities.Count, ex.Message), PxConstants.LOG_CATEGORY);
+				PxLoggingUtils.LogError(string.Format("Failed sending activities (count {0}) - {1}", activities.Count, ex.Message));
```
Comment (yaronpx) [inline PerimeterXModule/Internals/ActivityReporter.cs]: debug instead of error

File: PerimeterXModule/Internals/Helpers/PxLoggingUtils.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+		}
+
+		public static void LogDebug(String message)
+		{
+			Debug.WriteLine("[PerimeterX - DEBUG] [{0}] - {1}", m_appId, message);
+		}
+
+		public static void LogError(String message)
+		{
+			Debug.WriteLine("[PerimeterX - DEBUG] [{0}] - {1}", m_appId, message);
```
Comment (yaronpx) [inline PerimeterXModule/Internals/Helpers/PxLoggingUtils.cs]: should be `PerimeterX - ERROR`

File: PerimeterXModule/Internals/Validators/PXS2SValidator.cs (hunk 1 above)
Comment (alexbpx) [reply PerimeterXModule/Internals/Validators/PXS2SValidator.cs]: Even when logging an exception?

File: PerimeterXModule/PxModule.cs (hunk 2 above)
Comment (alexbpx) [reply PerimeterXModule/PxModule.cs]: fixed

File: PerimeterXModule/Internals/Helpers/PxLoggingUtils.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+		}
+
+		public static void LogDebug(String message)
+		{
+			Debug.WriteLine("[PerimeterX - DEBUG] [{0}] - {1}", m_appId, message);
+		}
+
+		public static void LogError(String message)
+		{
+			Debug.WriteLine("[PerimeterX - ERROR] [{0}] - {1}", m_appId, message);
```
Comment (yaronpx) [inline PerimeterXModule/Internals/Helpers/PxLoggingUtils.cs]: any way to make this write to error log?

File: PerimeterXModule/Internals/Validators/PXCookieValidator.cs [hunk 7]
```diff
@@ -114,7 +114,7 @@ public virtual bool Verify(PxContext context, IPxCookie pxCookie)
 			}
 			catch (Exception ex)
 			{
-				Debug.WriteLine("Request with invalid cookie (exception: " + ex.Message + ") - " + context.Uri, PxConstants.LOG_CATEGORY);
+				PxLoggingUtils.LogError("Request with invalid cookie (exception: " + ex.Message + ") - " + context.Uri);
```
Comment (yaronpx) [inline PerimeterXModule/Internals/Validators/PXCookieValidator.cs]: debug instead of error

File: PerimeterXModule/PxModule.cs [hunk 8]
```diff
@@ -222,7 +223,7 @@ private void Application_BeginRequest(object source, EventArgs e)
 			}
 			catch (Exception ex)
 			{
-				Debug.WriteLine("Failed to validate request: " + ex.Message, PxConstants.LOG_CATEGORY);
+				PxLoggingUtils.LogError("Failed to validate request: " + ex.Message);
```
Comment (yaronpx) [inline PerimeterXModule/PxModule.cs]: debug instead of error

**Pull request merged**